src/common/gui/SurgeGUIEditor.cpp

```diff
@@ -3751,6 +3753,47 @@ int32_t SurgeGUIEditor::controlModifierClicked(CControl* control, CButtonState b
                return 0;
             }
          }
+      } 
+      // exclusive mute/solo in the mixer
+      else if (button & kControl)
+      {
+         if (p->ctrltype == ct_bool_mute)
+         {
+            Parameter* o1 = &(synth->storage.getPatch().scene[current_scene].mute_o1);
+            Parameter* r23 = &(synth->storage.getPatch().scene[current_scene].mute_ring_23);
+
+            auto curr = o1;
+
+            while (curr <= r23)
+            {
+               if (curr->id == p->id)
+                  curr->val.b = true;
+               else
+                  curr->val.b = false;
+
+               curr++;
+            }
+         }
+         else if (p->ctrltype == ct_bool_solo)
+         {
+            Parameter* o1 = &(synth->storage.getPatch().scene[current_scene].solo_o1);
+            Parameter* r23 = &(synth->storage.getPatch().scene[current_scene].solo_ring_23);
+
+            auto curr = o1;
+
+            while (curr <= r23)
+            {
+               if (curr->id == p->id)
+                  curr->val.b = true;
+               else
+                  curr->val.b = false;
+
+               curr++;
+            }
+         }
+
+         synth->refresh_editor = true;
```

Comment:
We only need to do this refresh if our ctrltype is mute or solo. Can you move it inside the if blocks?